include/RED4ext/Types/SharedMutex.hpp

```diff
@@ -0,0 +1,29 @@
+#pragma once
+#include <memory>
+
+#include <Windows.h>
+
+#include <RED4ext/Common.hpp>
+
+namespace RED4ext
+{
+struct SharedMutex
+{
+    // -1 : write
+    // 0 : free
+    // + : read
+    volatile char state;
+
+    bool try_lock();
```

Comment:
snake_case, should be CamelCase

Reply:
named the functions like this to support std::lock_guard and std::shared_lock
do you want me to make it call "TryLock"? that way we have both PascalCase and the support of std::lock_guard

Reply:
I didn't think about that, I would say having both would be best, I am not sure if lock_guard and shared_lock support free functions, that would be best as that would mean not having the std functions in the class

Reply:
Just checked and it doesn't support free functions so I think having both versions would be the way to go :'(

Reply:
done